src/java/org/apache/cassandra/db/marshal/ListType.java

```diff
@@ -54,7 +54,7 @@ public static ListType<?> getInstance(TypeParser parser) throws ConfigurationExc
         if (l.size() != 1)
             throw new ConfigurationException("ListType takes exactly 1 type parameter");
 
-        return getInstance(l.get(0), true);
+        return getInstance(l.get(0).freeze(), true);
```

Comment:
It seems that freezing here is breaking `SSTableHeaderFixTest`

Reply:
yeah, I deferred that for the weekend... the type system now corrects the loss of freezing when parsing types, as this is enforced during the create table, but this impacts `SSTableHeaderFixTest` due to it not finding these types of columns in the fix set...

I am also wondering if we still want this logic as its 3.x -> 4.0 migration logic, and trunk is 5.0